data-prepper-pipeline-parser/src/main/java/org/opensearch/dataprepper/validation/PluginError.java

```diff
@@ -0,0 +1,42 @@
+package org.opensearch.dataprepper.validation;
+
+import lombok.Builder;
+import lombok.Getter;
+import lombok.NonNull;
+
+@Getter
+@Builder
+public class PluginError {
+    static final String PIPELINE_DELIMITER = ":";
+    static final String CAUSED_BY_DELIMITER = " caused by: ";
+    private final String pipelineName;
+    @NonNull
+    private final String pluginName;
+    @NonNull
+    private final Exception exception;
+
+    public String getErrorMessage() {
+        final StringBuilder message = new StringBuilder();
+        if (pipelineName != null) {
+            message.append(pipelineName);
+            message.append(PIPELINE_DELIMITER);
```

Comment:
nit: ERROR_MESSAGE_DELIMITER?

Reply:
This is between pipeline and plugin so that is why it is called pipeline DELIMITER.